### DISCUSSION TITLE
pydantic deprecation warning after updating to fastapi >= 0.111.0

### DISCUSSION THREAD

**Discussion by vincent-olivert-riera** [fastapi/fastapi#11921] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
I don't know where is the problem.
```

### Description

After updating from `fastapi[all]==0.110.3` to `fastapi[all]==0.111.0` (I use `pydantic==2.8.2` in both cases) the following warning happens when running the self tests of my application:

```
venv/lib/python3.11/site-packages/pydantic/_internal/_config.py:291:
  PydanticDeprecatedSince20: Support for class-based `config` is deprecated, use ConfigDict instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.8/migration/
```

This warning does not happen when using `fastapi[all]==0.110.3`. It started after the update to `fastapi[all]==0.111.0`. 

I have checked the URL provided in that warning message, but the instructions are for migrating from pydantic v1 to v2. I have always used v2, so that's not my case.

### Operating System

Linux

### FastAPI Version

Info in description

### Pydantic Version

2.8.2

### Python Version

3.11.2

**Comment by biwann13** [comment]
Hello, 
I've got the same issue with the following stack :
Python 3.11.1
Pydantic 2.9.2
FastAPI 0.115.0

Did you find a solution ?

**Reply by vincent-olivert-riera** [reply]
Unfortunately no, I didn't.

**Comment by hochstibe** [comment, marked as answer]
You dont't have any pydantic v1 model [configuration](https://docs.pydantic.dev/1.10/usage/model_config/) in your code?

```python
class Model(BaseModel):
    v: str

    class Config:
        max_anystr_length = 10
```

In Pydantic v2, you should use a [ConfigDict](https://docs.pydantic.dev/2.8/concepts/config/) instead of `class Config:`

```python
class Model(BaseModel):
    model_config = ConfigDict(str_max_length=10)

    v: str
```

**Reply by deanie77** [reply]
Thank you

**Reply by pintert3** [reply]
Funny, I don't seem to have any, but I still get this error. And I've checked all around for all these, and there's none

**Reply by vincent-olivert-riera** [reply]
For me, this was a conflict between FastAPI and pylenium. I didn't have any V1 model either. 

Once I stopped using pylenium the problem disappeared.
